src/io/flutter/sdk/FlutterSdkManager.java

```diff
@@ -5,71 +5,73 @@
 import com.intellij.openapi.roots.libraries.Library;
 import com.intellij.openapi.roots.libraries.LibraryTable;
+import com.intellij.openapi.util.Disposer;
 import com.intellij.util.EventDispatcher;
 import org.jetbrains.annotations.NotNull;
 
 import java.util.EventListener;
+import java.util.Timer;
+import java.util.TimerTask;
 
 /**
  * Monitors the application library table to notify clients when Flutter SDK configuration changes.
  */
 public class FlutterSdkManager {
   private final EventDispatcher<Listener> myDispatcher = EventDispatcher.create(Listener.class);
-  private final LibraryTableListener myLibraryTableListener = new LibraryTableListener();
-  private final RootProvider.RootSetChangedListener rootListener = x -> checkForFlutterSdkChange();
   private boolean isFlutterConfigured;
-  private final Project myProject;
 
   @NotNull
   public static FlutterSdkManager getInstance(@NotNull Project project) {
     return ServiceManager.getService(project, FlutterSdkManager.class);
   }
 
-  private FlutterSdkManager(Project project) {
-    myProject = project;
+  private FlutterSdkManager(@NotNull Project project) {
+    final LibraryTableListener libraryTableListener = new LibraryTableListener(project);
+    ProjectLibraryTable.getInstance(project).addListener(libraryTableListener);
 
-    listenForSdkChanges();
-    // Cache initial state.
-    isFlutterConfigured = isFlutterSdkSetAndNeeded();
-  }
+    final Timer timer = new Timer();
+    timer.scheduleAtFixedRate(new TimerTask() {
+      @Override
+      public void run() {
+        checkForFlutterSdkChange(project);
+      }
+    }, 1000, 1000);
 
-  private void listenForSdkChanges() {
-    ApplicationLibraryTable.getApplicationTable().addListener(myLibraryTableListener);
+    Disposer.register(project, () -> {
+      ProjectLibraryTable.getInstance(project).removeListener(libraryTableListener);
+      timer.cancel();
+    });
 
     ProjectManager.getInstance().addProjectManagerListener(new ProjectManagerAdapter() {
       @Override
       public void projectOpened(@NotNull Project project) {
-        checkForFlutterSdkChange();
+        checkForFlutterSdkChange(project);
       }
 
       @Override
       public void projectClosed(@NotNull Project project) {
-        checkForFlutterSdkChange();
+        checkForFlutterSdkChange(project);
       }
     });
 
-    // The Dart plugin modifies the library in place, so we need to listen for its root changes.
-    for (Library library : ApplicationLibraryTable.getApplicationTable().getLibraries()) {
-      watchDartSdkRoots(library);
-    }
+    // Cache initial state.
+    isFlutterConfigured = isFlutterSdkSetAndNeeded(project);
   }
 
   // Send events if Flutter SDK was configured or unconfigured.
-  public void checkForFlutterSdkChange() {
-    if (!isFlutterConfigured && isFlutterSdkSetAndNeeded()) {
+  public void checkForFlutterSdkChange(@NotNull Project project) {
```

Comment:
It seems like saving the Project to a field and removing the parameter would be better? (It eliminates the possibility of using a different project.)